addons/hw_drivers/tools/helpers.py

```diff
@@ -388,12 +421,21 @@ def load_iot_handlers():
                     _logger.exception('Unable to load handler file: %s', file)
     lazy_property.reset_all(http.root)
 
-def list_file_by_os(file_list):
-    platform_os = platform.system()
-    if platform_os == 'Linux':
-        return [x.name for x in Path(file_list).glob('*[!W].*')]
-    elif platform_os == 'Windows':
-        return [x.name for x in Path(file_list).glob('*[!L].*')]
+def list_file_by_iot_system(file_list: str) -> list:
+    """
+    List all handler files that an IoT system should load.
+    - IoT boxes load file without suffixe and _L
+    - Windows IoT load file without suffixes and _W
+    - Test IoT only load files with _T suffix
+    :param file_list: The path to the directory containing the files (either drivers or interfaces)
+    """
+    suffix_to_ignore = IoTSystem.get_all_value_except_me()
```

Comment:
if this is the only use of the `get_all_value_except_me` method, and it still needs to be stored in a variable (`suffix_to_ignore` here), then maybe we can't remove the method and compute it directly here. It will reduce the amount of code.

```suggestion
    suffix_to_ignore = IoTSystem.get_all_value() - IOT_SYSTEM.value
```